app/src/org/commcare/android/adapters/HomeScreenAdapter.java

```diff
@@ -0,0 +1,186 @@
+package org.commcare.android.adapters;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+
+import org.commcare.android.view.SquareButtonWithNotification;
+import org.commcare.dalvik.R;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+
+/**
+ * Created by dancluna on 3/19/15.
+ */
+public class HomeScreenAdapter extends BaseAdapter {
+    //region Buttons
+
+    static final int[] buttonsResources = new int[]{
+            R.layout.home_start_button,
+            R.layout.home_savedforms_button,
+            R.layout.home_incompleteforms_button,
+            R.layout.home_sync_button,
+            R.layout.home_disconnect_button,
+    };
+
+    static final HashMap<Integer, Integer> buttonsIDsToResources = new HashMap<Integer, Integer>() {{
+                put(R.id.home_start_sqbn,R.layout.home_start_button);
+                put(R.id.home_savedforms_sqbn,R.layout.home_savedforms_button);
+                put(R.id.home_sync_sqbn,R.layout.home_sync_button);
+                put(R.id.home_disconnect_sqbn,R.layout.home_disconnect_button);
+                put(R.id.home_incompleteforms_sqbn,R.layout.home_incompleteforms_button);
+    }};
+
+    //endregion
+
+    //region Private variables
+
+    final View.OnClickListener[] buttonListeners = new View.OnClickListener[buttonsResources.length];
```

Comment:
I'm curious why we need a buttonListeners array. Is it not possible to order the code in a way that ensures listeners are only added to existing buttons?